### PR TITLE
NOJIRA, expand scope of code linting (py, js)

### PR DESCRIPTION
**Pull request by johncrossman**
As promised, `flake8` covers all py files. Bonus: eslint rules unleashed.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -24,21 +24,22 @@ commands =
...
         --max-line-length 155 \
-        boac
+        boac config tests run.py
 deps =
     flake8
-    flake8-docstrings
+    flake8-commas
     flake8-import-order
+    flake8-pytest
+    flake8-quotes
```
Comment (johncrossman) [inline tox.ini]: Fyi, `flake8-docstrings` removed; no need for `D` rules.

File: tests/conftest.py [hunk 2]
```diff
@@ -12,8 +12,7 @@ def __init__(self, the_app, the_client):
 
     def login(self, uid):
         self.app.config['DEVELOPER_AUTH_ENABLED'] = True
-        self.client.post('/devauth/login',
-            data={'uid': uid, 'password': self.app.config['DEVELOPER_AUTH_PASSWORD']})
+        self.client.post('/devauth/login', data={'uid': uid, 'password': self.app.config['DEVELOPER_AUTH_PASSWORD']})
```
Comment (raydavis) [inline tests/conftest.py]: I don't understand this one, @johncrossman -- do you know what the motivation was?

File: config/default.py [hunk 3]
```diff
@@ -13,9 +13,9 @@
 
 # Some defaults.
 CSRF_ENABLED = True
-CSRF_SESSION_KEY = "secret"
+CSRF_SESSION_KEY = 'secret'
```
Comment (raydavis) [inline config/default.py]: Just curious -- was this flagged by a rule, or was it you as a human being consistent about the "textual matter in double-quotes" "computery strings in single" convention?

File: tests/conftest.py (hunk 2 above)
Comment (johncrossman) [reply tests/conftest.py]: @raydavis Violation was as below. I've amended PR to give it correct indentation.
```
E128 continuation line under-indented for visual indent
```

File: config/default.py (hunk 3 above)
Comment (johncrossman) [reply config/default.py]: @raydavis The robots asked for this change. Would you prefer to omit `config/` from py linting?

Comment (johncrossman) [reply config/default.py]: For the record, "Q000 Remove bad quotes" is the rule that was broken.

File: tests/conftest.py [hunk 4]
```diff
@@ -13,7 +13,7 @@ def __init__(self, the_app, the_client):
     def login(self, uid):
         self.app.config['DEVELOPER_AUTH_ENABLED'] = True
         self.client.post('/devauth/login',
-            data={'uid': uid, 'password': self.app.config['DEVELOPER_AUTH_PASSWORD']})
+                         data={'uid': uid, 'password': self.app.config['DEVELOPER_AUTH_PASSWORD']})
```
Comment (raydavis) [inline tests/conftest.py]: Ugh, personally I hate those super-long indents, and always turn off the formatters that default to them. @johncrossman , @paulkerschen , is there an alternative? PEP8 itself (as opposed to automated enforcers) just says there should be some visual distinction.

Maybe robocop would accept

```
        self.client.post(
            '/devauth/login',
            data={'uid': uid, 'password': self.app.config['DEVELOPER_AUTH_PASSWORD']}
        )
```

File: config/default.py (hunk 3 above)
Comment (raydavis) [reply config/default.py]: So we're basically deciding to centralize on single-quotes no matter what (except, I hope, when the text includes a single quote)? OK by me.

Comment (johncrossman) [reply config/default.py]: @raydavis Good news. As an example, Flake8 allows:
```
CANVAS_HTTP_TOKEN = "BOAC's secret"
```

File: tests/conftest.py (hunk 4 above)
Comment (paulkerschen) [reply tests/conftest.py]: @raydavis's compromise proposal makes sense to me, assuming the robot will play ball.

File: config/default.py (hunk 3 above)
Comment (paulkerschen) [reply config/default.py]: @johncrossman: so to be clear, the rules to keep the robots happy are:
- Double quotes for thrice-quoted docstrings;
- Single quotes for other strings;
- Unless the string contains a single quote, in which case double quotes may e used;
- If your string contains both single and double quotes, you've been at work too long and need to go home;
?

File: tests/conftest.py (hunk 4 above)
Comment (johncrossman) [reply tests/conftest.py]: @raydavis @paulkerschen Robocop seems to like:
```
self.client.post(
    '/devauth/login',
    data={'uid': uid, 'password': self.app.config['DEVELOPER_AUTH_PASSWORD']},
)
```
Works for me. Commit amended.

File: config/default.py (hunk 3 above)
Comment (johncrossman) [reply config/default.py]: @paulkerschen That's right. Note: If the string contains single quote(s) then there is the option of escaping them. For example :
```
CANVAS_HTTP_TOKEN = 'BOAC\'s secret'
```
The robots are flexible. (I prefer escaping, as shown above, over the _sometimes-we-use-double-quotes_ principle. But I, too, am flexible.)   
cc: @raydavis

**Comment by paulkerschen** [comment]
Thanks for this, @johncrossman; I make bold to declare that coexistence with the robots is possible.

**Pull request merged**